### PR TITLE
Initial draft of mysql full provisioner 

### PR DESCRIPTION
**Pull request by nothingmn**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unit Tests

Optional:

- [ ] Functional Tests
- [ ] Integration Tests

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/mysqld/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
+# Change Log
```
Comment (nsainaney) [inline packages/mysqld/CHANGELOG.md:1]: I'd delete this file - it's autogenerated on publish

File: packages/mysqld/c6o/app.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: Mysql (App)
```
Comment (nsainaney) [inline packages/mysqld/c6o/app.yaml:1]:
```suggestion
name: MySql
```

File: packages/mysqld/src/mixins/createValidate.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { baseProvisionerType } from '../'
+
+export const createValidateMixin = (base: baseProvisionerType) => class extends base {
+
+    async createValidate() {
+        // TODO: check fields are all set
+        this.spec.secretKeyRef = this.spec.secretKeyRef || 'mongo-connections'
```
Comment (nsainaney) [inline packages/mysqld/src/mixins/createValidate.ts:7]: Defaults to mongo-connections

File: packages/mysqld/k8s/statefulset.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+apiVersion: apps/v1
```
Comment (nsainaney) [inline packages/mysqld/k8s/statefulset.yaml:1]: Should this be a stateful set? How do the MySQL pods discover each other?

**Review by nsainaney — CHANGES_REQUESTED** [review]

File: packages/mysqld/k8s/statefulset.yaml (hunk 4 above)
Comment (nothingmn) [reply packages/mysqld/k8s/statefulset.yaml:1]: Ill remove the stateful set for now.

**Pull request closed without merging**